### ISSUE TITLE
Feature: use outlet-links in size|position params

### ISSUE THREAD

**Issue by comm1x** [schibsted/layout#54]
I suggest to use outlets in size|position-params. For example:

```
<UIView
    outlet="topLeftView"
    top="0"
    left="0" />

<UILabel
    outlet="view2"
    top="topLeftView.bottom"           <--- Link to the position of another view
    left="topLeftView.right" />

<MyCustomButton
    top="view2.top + 20"
    left="view2.left + 20" />
```

I think, it is not trivial, because we need to calculate view graph (hierarchy, that is not equal xml-hierarchy) and detect cycles.
But if it is implemented, the Layout become **very powerful** and will be able to completely replace IB when building complex interfaces with complex view-constraints.

**Comment by nicklockwood** [comment]
Interesting idea! I was originally planning to do something more like this though:

```xml
<UIView
    id="topLeftView"
    top=""
    left="0" />

<UILabel
    outlet="view2"
    top="#topLeftView.bottom"           <--- Link to the position of another view
    left="#topLeftView.right" />
```

The problem with using outlets for this is that they are designed to bind to the file's owner (view controller), and you might want to have references between views that aren't bound to the view controller.

I did have this implemented at one point, so I know it's possible. Cycles aren't a problem as I already have cycle detection, but it does make it harder to limit the scope of updates.

**Comment by comm1x** [comment]
I am not sure about make separate `outlet` and `id` fields. In Kotlin (kotlin-android-extensions) this fields combined into `id`.

For example:
```
<Button android:id="@+id/leftButton" ... />
<TextView android:id="@+id/title" android:layout_toRightOf="@+id/leftButton" ... />
```

And in code:
```
class Toolbar(context: Context, attrs: AttributeSet?) : RelativeLayout(context, attrs) {
    init {
        View.inflate(context, R.layout.toolbar, this)
        // imlicit calling findViewById(R.id.title) as TextView
        // binded name same as ID
        title.text = "qwe"              
        leftButton.setOnClickListener { doSomething() }
```

If we split `id` and `outlet`, I think xml-layouts will looks like:
```
<UIView
    id="container"
   outlet="container" ... />
```

Id and outlet will offen duplicates.

**Comment by nicklockwood** [comment]
@comm1x FYI, I've put up a PR with the "id"-based solution, so you may wish to comment on it: https://github.com/schibsted/layout/pull/61

I'm aware of your concerns about duplicating the id and outlet. I considered making the #references work with outlets as well as id, but I'm still not convinced that the doubling up you describe would be very common, since the two properties serve such different purposes.

It won't be hard to revisit this decision later though, so if after using the feature you find that this comes up a lot I'd encourage you to raise it again.

**Comment by nicklockwood** [comment]
This feature is now available in 0.5.4

**Comment by comm1x** [comment]
I am still collecting statistic about my id/outlet usage. 
But right now I already have exactly 100% of duplicating. All 8 of my 8 cases have the same `id` and `outlet`. Not so many, the sample is not representative, but I'm already confident that the percentage of duplication will be very high.

The main premise of duplication is the absence of a reason not to do so.
